CMakeLists.txt

```diff
@@ -0,0 +1,97 @@
+# Minimum CMake required
+cmake_minimum_required(VERSION 3.1)
+
+# Project
+project(onnxir C CXX)
+
+# Set C++11 as standard for the whole project
+set(CMAKE_CXX_STANDARD 11)
+
+set(ONNXIR_ROOT ${PROJECT_SOURCE_DIR})
+if (WIN32)
+    # parallel build
+    add_compile_options(/MP)
+    add_compile_options(/WX)
+    set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /WX")
+    set(CMAKE_STATIC_LINKER_FLAGS "${CMAKE_STATIC_LINKER_FLAGS} /WX")
+endif()
+add_subdirectory(third_party/pybind11)
+FIND_PACKAGE(Protobuf REQUIRED)
+include_directories(
+    ${ONNXIR_ROOT}
+    ${CMAKE_CURRENT_BINARY_DIR}
+    # External dependencies.
+    ${PROTOBUF_INCLUDE_DIRS}
+    ${PYBIND11_INCLUDE_DIR}
+    ${PYTHON_INCLUDE_DIRS}
+)
+
+function(RELATIVE_PROTOBUF_GENERATE_CPP SRCS HDRS ROOT_DIR)
```

Comment:
We cribbed this function definition from somewhere, right? Let's record its provenance please.

Reply:
Aha, that's sth from ms code base, let me ask people whether they know where they copied from.

Reply:
looks like I can't figure out the code owner for now. I searched around and https://github.com/tensorflow/tensorflow/blob/d2c3b873c6f8ff999a2e4ee707a84ff00d9c15a5/tensorflow/contrib/cmake/tf_core_framework.cmake has the most similar one, so I'm going to put this link as a comment in place, sounds good?